R/make_datatagr.R

```diff
@@ -1,21 +1,18 @@
 #' Create a datatagr from a data.frame
 #'
 #' This function converts a `data.frame` or a `tibble` into a `datatagr` object,
-#' where different types of epidemiologically relevant data are tagged. This
-#' includes dates of different events (e.g. onset of symptoms, case reporting),
-#' information on the patient (e.g. age, gender, location) as well as other
-#' information such as the type of case (e.g. confirmed, probable) or the
-#' outcome of the disease. The output will seem to be the same `data.frame`, but
+#' where data are tagged. The output will seem to be the same `data.frame`, but
 #' `datatagr`-aware packages will then be able to automatically use tagged
 #' fields for further data cleaning and analysis.
 #'
-#' @param x a `data.frame` or a `tibble` containing case line list data, with
-#'   cases in rows and variables in columns
+#' @param x a `data.frame` or a `tibble`
 #'
 #' @param ... <[`dynamic-dots`][rlang::dyn-dots]> A series of tags provided as
-#'   `tag_name = "column_name"`, where `tag_name` indicates any of the known
-#'   variables listed in 'Details' and values indicate their name in `x`; see
-#'   details for a list of known variable types and their expected content
+#'   `tag_name = "column_name"`. When specifying tags, please also see
+#'   `tag_defaults` to specify default values.
+#'
+#' @param tag_defaults a list of default values for the provided tags. Defaults
+#'   to `list()`, effectively defaulting to NULL values.
```

Comment:
I struggle a little bit to understand the context in this would be used. My first intuition would have been to remove this entirely.

Can you add an example with this argument please?

Reply:
Yes that is right. 

I removed the **linelist** specific `tags_defaults.R` and was thinking about how to keep **datatagr** compatible. In other words, I do not see a need for this argument except for linelist reasons.

Revisiting this in **linelist** indicates it is primarily related to the types of the tags:

```
tags_defaults <- function() {
  setNames(
    vector(
      "list",
      length = length(tags_types())
    ),
    names(tags_types())
  )
}
```

As a result, how about I remove the argument `tag_defaults` and instead allow users to supply the `tag_types` directly when making the object? This will then be backwards compatible with linelist and allow for us to implement immediate validation in the creation of the object, instead of only upon calling a separate validation function (e.g., `validate_datatagr` or `validate_types`)?

Reply:
That sounds good – I am a big proponent of practical over hypothetical solutions 😊 Started tracking the overall problem of linelist compatibility points in #25.